File: newMyDataStructureAlgorithm/algorithm/combination.cpp
//https://www.acmicpc.net/problem/15650

#include <cstdio>

int a[10];
int N, M;
void dfs(int depth, int start)
{
	if (depth == M + 1)
	{
		for (int i = 1; i <= M; i++) {
			printf("%d ", a[i]);
		}
		printf("\n");
		return;
	}
	for (int i = start; i <= N; i++)
	{
		a[depth] = i;
		dfs(depth + 1, i + 1);
	}
}

int main() {
	scanf_s("%d %d", &N, &M);
	dfs(1, 1);
}

File: newMyDataStructureAlgorithm/algorithm/combination.js
//https://www.acmicpc.net/problem/15650
let input = require('fs').readFileSync('/dev/stdin').toString().trim().split(' ');

let N = +(input[0].trim());
let M = +(input[1].trim());
let array = [];

const combination =(index, start, n, m) => {
  if (index == m) {
    console.log(array.join(' '));
    return
  }

  for (let i = start; i <= n; i++) {
    array[index] = i;
    combination(index + 1, i + 1, n, m);
  }
};

combination(0, 1, N, M);
console.log(combination(0, 1, 4 ,2));